File: src/NocWMS-Server/Server.ApiService/Models/InboundReceipt.cs
using System;
using System.Collections.Generic;
using KiraNet.Camellia.Infrastructure.DomainModel.Data;

namespace Server.ApiService.Models
{
    public class InboundReceipt : IEntity
    {
        public string Id { get; set; }
        public int WarehouseId { get; set; }
        public string VendorId { get; set; }
        public string WaybillNo { get; set; }
        public string StaffId { get; set; }
        public string HandlerName { get; set; }
        public string Acceptor { get; set; }
        public string Deliveryman { get; set; }
        public bool IsDone { get; set; }
        public DateTime CreateTime { get; set; }

        public Warehouse Warehouse { get; set; }
        public Vendor Vendor { get; set; }
        public Staff Staff { get; set; }
        public List<InboundReceiptItem> InboundReceiptItems { get; set; }
    }
}

File: src/NocWMS-Client/src/app/models/inbound-receipt.ts
import { InboundReceiptItem } from "./inbound-receipt-item";

export class InboundReceipt {
    constructor(inboundReceipt?: Partial<InboundReceipt>) {
        if (inboundReceipt) {
            this.id = inboundReceipt.id;
            this.staffId = inboundReceipt.staffId;
            this.warehouseId = inboundReceipt.warehouseId;
            this.warehouseName = inboundReceipt.warehouseName;
            this.vendorId = inboundReceipt.vendorId;
            this.vendorName = inboundReceipt.vendorName;
            this.handlerName = inboundReceipt.handlerName;
            this.acceptor = inboundReceipt.acceptor;
            this.deliveryman = inboundReceipt.deliveryman
            this.waybillNo = inboundReceipt.waybillNo;
            this.isDone = inboundReceipt.isDone;
            this.createTime = inboundReceipt.createTime;
            this.inboundReceiptItems = inboundReceipt.inboundReceiptItems;
        }
    }

    public id: string;
    public staffId: string;
    public warehouseId: number;
    public warehouseName: string;
    public vendorId: string;
    public vendorName: string;
    public handlerName: string;
    public acceptor: string;
    public deliveryman: string;
    public waybillNo: string;
    public isDone: boolean;
    public createTime: string;
    public inboundReceiptItems: InboundReceiptItem[];
    public get total(): number {
        let sum = 0;
        if (this.inboundReceiptItems) {
            this.inboundReceiptItems.forEach((v) => sum += v.productPrice * v.num);
        }

        return sum;
    }
}